### PR TITLE
Strict deps resolution else fail fast

### PR DESCRIPTION
**Pull request by schellap**
When assets in the deps are not found, fail fast.

Only when deps file is missing add the assemblies in the app base.

### CODE REVIEW

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 1]
```diff
@@ -329,7 +329,6 @@ public static BuildTargetResult CompileStage1(BuildTargetContext c)
                 dotnet: DotNetCli.Stage0,
                 outputDir: Dirs.Stage1);
 
-            CleanOutputDir(Path.Combine(Dirs.Stage1, "sdk"));
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Looks like the method `CleanOutputDir` can now be removed as well. And also the `FilesToClean` static array.

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 2]
```diff
@@ -657,7 +655,6 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
                 {
                     // Yes, it is .exe even on Linux. This is the managed exe we're working with
                     File.Copy(Path.Combine(binaryToCorehostifyOutDir, $"{binaryToCorehostify}.exe"), Path.Combine(outputDir, $"{binaryToCorehostify}.dll"));
-                    File.Delete(Path.Combine(binaryToCorehostifyOutDir, $"{binaryToCorehostify}.exe"));
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/CompileTargets.cs]: ?? I think we should continue deleting the `csc.exe` file, since it isn't being used - we don't want it in our SDK folder.

Comment (schellap) [inline scripts/dotnet-cli-build/CompileTargets.cs]: If an assembly specified by the deps file is not found, I am flagging an error.

So I had to remove this deletion. The other option is to remove this entry from the deps file as well. At that point, it is editing a project.lock.json. Both of these are not pretty, do you have other ideas or opinions? @piotrpMSFT.

Comment (eerhardt) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Maybe we shouldn't be renaming `csc.exe` to `csc.dll` at all?  Maybe we should just leave it `csc.exe` - just like we are for `vbc.exe` now.

/cc @pakrym who added this code.

Comment (pakrym) [inline scripts/dotnet-cli-build/CompileTargets.cs]: I've added renaming code because `csc.exe` could not be ran directly with portable cli setup. Having two csc's does not make sense, but you can change `ChangeEntryPointLibraryName` second argument to null and it will delete deps entry completely.

**Comment by schellap** [comment]
Adding a note: test sqlclient, musicstore.

**Comment by schellap** [comment]
@gkhanna79 PTAL. In commit 2, I am retaining mscorlib.dll in the Shared FX. NI images can be incomplete right and may need the IL? If not, I can undo commit 2 and not flag an error for mscorlib, because coreclr anyways would error out if it can't find it.

@eerhardt I manually removed the entries from the `runtimeTargets` section.

**Comment by gkhanna79** [comment]
> Only when deps file is missing add the assemblies in the app base.

Shouldn't this say "only when app is standalone..."?

File: build_projects/dotnet-cli-build/CompileTargets.cs [hunk 3]
```diff
@@ -318,13 +318,6 @@ public static void PublishSharedFramework(BuildTargetContext c, string outputDir
                 Path.Combine(Dirs.CorehostLatest, HostPolicyBaseName),
                 Path.Combine(SharedFrameworkNameAndVersionRoot, HostPolicyBaseName), true);
 
-            if (File.Exists(Path.Combine(SharedFrameworkNameAndVersionRoot, "mscorlib.ni.dll")))
```
Comment (gkhanna79) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: NI contains the IL. We should undo this change.

Comment (schellap) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: > Shouldn't this say "only when app is standalone..."?

No, this is for portable too without rid assets for app dependency assemblies.

Comment (schellap) [inline build_projects/dotnet-cli-build/CompileTargets.cs]: > NI contains the IL

I thought only R2R contained them. Thanks for clarifying. I'm leaving the error handling for mscorlib to CoreCLR since apps can choose to delete one or the other.

**Comment by schellap** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by schellap** [comment]
@dotnet-bot test this please

**Comment by schellap** [comment]
@dotnet-bot test this please

**Comment by schellap** [comment]
@dotnet-bot test this please

**Comment by schellap** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by schellap** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by schellap** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by schellap** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by schellap** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please

**Comment by schellap** [comment]
@dotnet-bot test this please
